src/seedu/addressbook/AddressBook.java

```diff
@@ -608,8 +609,10 @@ private static String executeClearAddressBook() {
      */
     private static String executeUpdatePerson(String commandArgs)   {
         final String[] updateStringSplit = splitIndexAndPersonData(commandArgs);
+        final Optional<String[]> decodeResult = decodePersonFromString(updateStringSplit[1]);
 
-        if (!isDeletePersonArgsValid(updateStringSplit[0])) {
+        // checks if args are valid (decode result will not be present if the person is invalid)
+        if (!isDeletePersonArgsValid(updateStringSplit[0]) || !decodeResult.isPresent()) {
```

Comment:
Similar to the previous comment, you might want to rename the _isDeletePersonArgsValid_ method, so that it makes more sense for someone else who is reading the code subsequently.